azkaban-common/src/main/java/azkaban/executor/ExecutionController.java

```diff
@@ -689,8 +689,9 @@ public String submitExecutableFlow(final ExecutableFlow exflow, final String use
             this.maxConcurrentRunsPerFlowMap);
         if (running.size() > maxConcurrentRuns) {
           this.commonMetrics.markSubmitFlowSkip();
-          throw new ExecutorManagerException("Flow " + flowId
-              + " has more than " + maxConcurrentRuns + " concurrent runs. Skipping",
+          throw new ExecutorManagerException("Flow with id " +
+              (exflow.getFlowDefinitionId() > 0 ? exflow.getFlowDefinitionId() : flowId)
```

Comment:
Is the `flowDefinitionId` sufficient to uniquely identify the flow or does it need to be the tuple `<flowId,flowDefinitionId>` ?

Reply:
This is to maintain compatibility with existent code. `flowDefinitionId` will only have a value when creating executions via the new apis.